cocotb/share/lib/embed/gpi_embed.c

```diff
@@ -384,3 +384,10 @@ void embed_sim_event(gpi_event_t level, const char *msg)
 
     FEXIT
 }
+
+void embed_sim_end(void)
+{
+    FENTER
+
+    Py_Finalize();
```

Comment:
personally I think this FENTER / FEXIT stuff is garbage, but if we're going to use it, you should add an FEXIT here too

Reply:
I was under the impression that the gpi_log used python's logger. If the interpreter is finalized we can't log anything thereafter.